lib/spec.go

```diff
@@ -44,6 +45,40 @@ type Mock struct {
 	calls int
 }
 
+// Mocks metadata such as workspace
+type MockMetadata struct {
+	Workspace string
+}
+
+type assert struct {
+	Type      string         `hcl:"type,label"`
+	Name      string         `hcl:"name,label"`
+	Config    hcl.Body       `hcl:",remain"`
+	DependsOn hcl.Expression `hcl:"depends_on,attr"`
+}
+
+type mock struct {
+	Type   string   `hcl:"type,label"`
+	Name   string   `hcl:"name,label"`
+	Config hcl.Body `hcl:",remain"`
+}
+
+type mock_metadata struct {
+	Config hcl.Body `hcl:",remain"`
+}
+
+type MetadataConfig struct {
+	Workspace string `hcl:"workspace"`
+}
+
+type root struct {
```

Comment:
Unfortunately because it makes sense to have the metadata mock per file and how things are passed in we need two methods to be able to read the struct. Not sure how you are envisioning how this would work. I just did the easiest thing and pull it out of the method.